java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/ArrowVectorIterator.java

```diff
@@ -134,11 +122,26 @@ private VectorSchemaRoot createVectorSchemaRoot() {
       }
       throw new RuntimeException("Error occurred while creating schema root.", e);
     }
+
+    // ensure consumers have been initialized
+    ensureInitialized(root);
     return root;
   }
 
+  private void ensureInitialized(VectorSchemaRoot root) throws SQLException {
+    if (!initialized) {
```

Comment:
It seems this isn't right if `!config.isReuseVectorSchemaRoot()` because we'll have recreated the root. Since `ensureInitialized` is only ever called when creating a new root, I don't think we need to guard this with `initialized`?

Reply:
Ah yes good point!

Reply:
Actually, it's fine as its currently written. After creating a new `VectorSchemaRoot` we'll call `load` and this will call `JdbcConsumer#resetValueVector` passing the vectors of the _new_ `VectorSchemaRoot` and obtaining a reference to them (https://github.com/apache/arrow/pull/12692/files#diff-f812c76a565e7c56500943f512b8498487209b15ed036d404d703854841df3d0R161). This will happen before we consume data.

Reply:
Actually sorry I see we have to recreate the delegate consumer as it references the child vector so it does make sense to initialize each time.